### DISCUSSION TITLE
Index based confgurations clarifcation

### DISCUSSION THREAD

**Discussion by henkepa** [rabbitmq/rabbitmq-server#7251] [Questions] [answered]
Dears,

in cases with index based configurations like

################################################################
cluster_formation.classic_config.nodes.**1** = rabbit@hostname1.eng.example.local
cluster_formation.classic_config.nodes.**2** = rabbit@hostname2.eng.example.local
################################################################

Do these indices need to be numeric or is it sufficient for them to be unique?
If they must be numeric, may they also start with 0?

**Comment by henkepa** [comment]
@michaelklishin are you able to help here? That would be highly appreciated

**Comment by gomoripeti** [comment, marked as answer]
hi @henkepa , the indices can be anything, do not need to be numbers. They are ignored by the config parser
(For details, this is where parsing this particular key is implemented https://github.com/rabbitmq/rabbitmq-server/blob/v3.11.9/deps/rabbit/priv/schema/rabbit.schema#L1301)

**Reply by henkepa** [reply]
thanks a lot

**Comment by michaelklishin** [comment]
@gomoripeti is correct, it is sufficient for them to be unique in most cases. In some rare cases the values of the keys themselves can be used but in the example above, they are ignored and only the values are retained. That said, if they were not unique the config parser would likely end up with a collection of just one node.
